--- profile-dotemacs.el	2025-06-06 14:13:37.534701122 -0400
+++ safe/profile-dotemacs.el	2025-02-23 15:06:59.003251569 -0500
@@ -1,4 +1,4 @@
-;;; profile-dotemacs.el --- Profile your Emacs init file
+;;; profile-dotemacs.el --- Profile your Emacs init file -*- lexical-binding: t -*-
 
 ;; Copyright (C) 2010, 2012  David Engster
 
@@ -49,7 +49,7 @@
 
 ;; Start emacs as follows:
 ;;
-;;    emacs -Q -l <PATH>/profile-dotemacs.el -f profile-dotemacs
+;;    emacs -Q -l ~/.emacs.d/extern/profile-dotemacs.el -f profile-dotemacs
 ;;
 ;; with <PATH> being the path to where this file resides.
 
@@ -77,7 +77,7 @@
 
 ;; User variables
 
-(defvar profile-dotemacs-file "~/.emacs"
+(defvar profile-dotemacs-file "~/.emacs.d/init.el"
   "File to be profiled.")
 
 (defvar profile-dotemacs-low-percentage 3
@@ -87,17 +87,17 @@
 
 (defface profile-dotemacs-time-face
   '((((background dark)) (:background "OrangeRed1"))
-    (t (:background "red3")))
+     (t (:background "red3")))
   "Background color to indicate percentage of total time.")
 
 (defface profile-dotemacs-low-percentage-face
   '((((background dark)) (:foreground "gray25"))
-    (t (:foreground "gray75")))
+     (t (:foreground "gray75")))
   "Face for sexps below `profile-dotemacs-low-percentage'.")
 
 (defface profile-dotemacs-highlight-face
   '((((background dark)) (:background "blue"))
-    (t (:background "yellow")))
+     (t (:background "yellow")))
   "Highlight face for benchmark results.")
 
 ;; Main function
@@ -110,17 +110,17 @@
     (goto-char (point-min))
     (let (start end results)
       (while
-	  (< (point)
-	     (setq end (progn
-			 (forward-sexp 1)
-			 (point))))
+	(< (point)
+	  (setq end (progn
+		      (forward-sexp 1)
+		      (point))))
 	(forward-sexp -1)
 	(setq start (point))
 	(add-to-list
-	 'results
-	 `(,start ,end
-		  ,(benchmark-run
-		    (eval (sexp-at-point)))))
+	  'results
+	  `(,start ,end
+	     ,(benchmark-run
+		(eval (sexp-at-point)))))
 	(goto-char end))
       (profile-dotemacs-show-results results)
       (switch-to-buffer (current-buffer)))))
@@ -130,36 +130,36 @@
 (defun profile-dotemacs-show-results (results)
   "Show timings from RESULTS in current buffer."
   (let ((totaltime (profile-dotemacs-totaltime results))
-	current percentage ov)
+	 current percentage ov)
     (while results
       (let* ((current (pop results))
-	     (ov (make-overlay (car current) (cadr current)))
-	     (current (car (last current)))
-	     (percentage (/ (+ (car current) (nth 2 current))
+	      (ov (make-overlay (car current) (cadr current)))
+	      (current (car (last current)))
+	      (percentage (/ (+ (car current) (nth 2 current))
 			    totaltime))
-	     col benchstr lowface)
+	      col benchstr lowface)
 	(setq col
-	      (profile-dotemacs-percentage-color
-	       percentage
-	       (face-background 'default)
-	       (face-background 'profile-dotemacs-time-face)))
+	  (profile-dotemacs-percentage-color
+	    percentage
+	    (face-background 'default)
+	    (face-background 'profile-dotemacs-time-face)))
 	(setq percentage (round (* 100 percentage)))
 	(setq benchstr (profile-dotemacs-make-benchstr current))
 	(overlay-put ov 'help-echo benchstr)
 	(if (and (numberp profile-dotemacs-low-percentage)
-		 (< percentage profile-dotemacs-low-percentage))
-	    (overlay-put ov 'face 'profile-dotemacs-low-percentage-face)
+	      (< percentage profile-dotemacs-low-percentage))
+	  (overlay-put ov 'face 'profile-dotemacs-low-percentage-face)
 	  (overlay-put ov 'before-string
-		       (propertize benchstr
-				   'face 'profile-dotemacs-highlight-face))
+	    (propertize benchstr
+	      'face 'profile-dotemacs-highlight-face))
 	  (overlay-put ov 'face
-		       `(:background ,col)))))
+	    `(:background ,col)))))
     (setq ov (make-overlay (1- (point-max)) (point-max)))
     (overlay-put ov 'after-string
-		 (propertize
-		  (format "\n-----------------\nTotal time: %.2fs\n"
-			  totaltime)
-		  'face 'profile-dotemacs-highlight-face))))
+      (propertize
+	(format "\n-----------------\nTotal time: %.2fs\n"
+	  totaltime)
+	'face 'profile-dotemacs-highlight-face))))
 
 (defun profile-dotemacs-totaltime (results)
   "Calculate total time of RESULTS."
@@ -167,34 +167,34 @@
     (mapc (lambda (x)
 	    (let ((cur (car (last x))))
 	      (setq totaltime (+ totaltime (car cur) (nth 2 cur)))))
-	  results)
+      results)
     totaltime))
 
 (defun profile-dotemacs-percentage-color (percent col-begin col-end)
   "Calculate color according to PERCENT between COL-BEGIN and COL-END."
   (let* ((col1 (color-values col-begin))
-	 (col2 (color-values col-end))
-	 (col
-	  (mapcar (lambda (c)
-		    (round
-		     (+ (* (- 1 percent) (nth c col1))
-			(* percent (nth c col2)))))
-		  '(0 1 2))))
+	  (col2 (color-values col-end))
+	  (col
+	    (mapcar (lambda (c)
+		      (round
+		        (+ (* (- 1 percent) (nth c col1))
+			  (* percent (nth c col2)))))
+	      '(0 1 2))))
     (format "RGB:%04x/%04x/%04x"
-	    (car col)
-	    (nth 1 col)
-	    (nth 2 col))))
+      (car col)
+      (nth 1 col)
+      (nth 2 col))))
 
 (defun profile-dotemacs-make-benchstr (timings)
   "Create descriptive benchmark string from TIMINGS."
   (format
-   (concat
-    "<Percentage: %d ; "
-    "Time: %.2f ; "
-    "Number of GC: %d ; "
-    "Time for GC: %.2f>\n")
-   percentage
-   (car timings) (nth 1 timings) (nth 2 timings)))
+    (concat
+      "<Percentage: %d ; "
+      "Time: %.2f ; "
+      "Number of GC: %d ; "
+      "Time for GC: %.2f>\n")
+    percentage
+    (car timings) (nth 1 timings) (nth 2 timings)))
 
 
 ;; profile-dotemacs.el ends here
